state/pruner.go

```diff
@@ -81,6 +92,7 @@ func (p *Pruner) SetObserver(obs PrunerObserver) {
 
 func (p *Pruner) OnStart() error {
 	go p.pruningRoutine()
+	go p.indexerPruningRoutine()
```

Comment:
great job! This will make sure the indexer pruning is not in the way of the other pruning processes.